commit 3f28cee7ffa83df4510b234af13c223714b9324c
Author: darin@apple.com <darin@apple.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date:   Fri Feb 25 02:20:42 2011 +0000

    2011-02-24  Darin Adler  <darin@apple.com>
    
            Reviewed by Alexey Proskuryakov.
    
            REGRESSION (r79466): http/tests/incremental/slow-utf8-html.pl flaky due to incorrect assertions
            https://bugs.webkit.org/show_bug.cgi?id=55135
    
            * platform/text/TextCodecUTF8.cpp:
            (WebCore::TextCodecUTF8::decode): Removed incorrect assertions.
    
    
    git-svn-id: http://svn.webkit.org/repository/webkit/trunk@79655 268f45cc-cd09-0410-ab3c-d52691b4dbfc

diff --git a/Source/WebCore/platform/text/TextCodecUTF8.cpp b/Source/WebCore/platform/text/TextCodecUTF8.cpp
index 29426ad..5f82092 100644
--- a/Source/WebCore/platform/text/TextCodecUTF8.cpp
+++ b/Source/WebCore/platform/text/TextCodecUTF8.cpp
@@ -227,18 +227,16 @@ String TextCodecUTF8::decode(const char* bytes, size_t length, bool flush, bool
 
     do {
         if (m_partialSequenceSize) {
-            // Explicitly copy destination and source to avoid taking a pointer to them,
-            // which may harm code generation.
+            // Explicitly copy destination and source pointers to avoid taking pointers to the
+            // local variables, which may harm code generation by disabling some optimizations
+            // in some compilers.
             UChar* destinationForHandlePartialSequence = destination;
             const uint8_t* sourceForHandlePartialSequence = source;
             handlePartialSequence(destinationForHandlePartialSequence, sourceForHandlePartialSequence, end, flush, stopOnError, sawError);
             destination = destinationForHandlePartialSequence;
             source = sourceForHandlePartialSequence;
-            if (m_partialSequenceSize) {
-                ASSERT(stopOnError);
-                ASSERT(sawError);
+            if (m_partialSequenceSize)
                 break;
-            }
         }
 
         while (source < end) {
